src/raw.rs

```diff
@@ -0,0 +1,431 @@
+use std::ops::RangeBounds;
+
+use futures::{Future, Poll};
+
+use {Config, Key, KvFuture, KvPair, Value};
+
+#[derive(Default, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct ColumnFamily(String);
+
+impl Into<ColumnFamily> for String {
+    fn into(self) -> ColumnFamily {
+        ColumnFamily(self)
+    }
+}
+
+impl<'a> Into<ColumnFamily> for &'a str {
+    fn into(self) -> ColumnFamily {
+        ColumnFamily(self.to_owned())
+    }
+}
+
+pub struct Get<'a, AClient: Client> {
+    client: &'a AClient,
+    key: Key,
+    cf: Option<ColumnFamily>,
+}
+
+impl<'a, AClient: Client> Get<'a, AClient> {
+    fn new(client: &'a AClient, key: Key) -> Self {
+        Get {
+            client,
+            key,
+            cf: None,
+        }
+    }
+
+    pub fn cf(mut self, cf: impl Into<ColumnFamily>) -> Self {
+        self.cf = Some(cf.into());
+        self
+    }
+}
+
+impl<'a, AClient: Client> Future for Get<'a, AClient> {
+    type Item = Value;
+    type Error = ();
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        let _ = &self.client;
+        let _ = &self.key;
+        let _ = &self.cf;
+        unimplemented!()
+    }
+}
+
+pub struct BatchGet<'a, AClient: Client> {
+    client: &'a AClient,
+    keys: Vec<Key>,
+    cf: Option<ColumnFamily>,
+}
+
+impl<'a, AClient: Client> BatchGet<'a, AClient> {
+    fn new(client: &'a AClient, keys: Vec<Key>) -> Self {
+        BatchGet {
+            client,
+            keys,
+            cf: None,
+        }
+    }
+
+    pub fn cf(mut self, cf: impl Into<ColumnFamily>) -> Self {
+        self.cf = Some(cf.into());
+        self
+    }
+}
+
+impl<'a, AClient: Client> Future for BatchGet<'a, AClient> {
+    type Item = Vec<KvPair>;
+    type Error = ();
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        let _ = &self.client;
+        let _ = &self.keys;
+        let _ = &self.cf;
+        unimplemented!()
+    }
+}
+
+pub struct Put<'a, AClient: Client> {
+    client: &'a AClient,
+    pair: KvPair,
+    cf: Option<ColumnFamily>,
+}
+
+impl<'a, AClient: Client> Put<'a, AClient> {
+    fn new(client: &'a AClient, pair: KvPair) -> Self {
+        Put {
+            client,
+            pair,
+            cf: None,
+        }
+    }
+
+    pub fn cf(mut self, cf: impl Into<ColumnFamily>) -> Self {
+        self.cf = Some(cf.into());
+        self
+    }
+}
+
+impl<'a, AClient: Client> Future for Put<'a, AClient> {
+    type Item = ();
+    type Error = ();
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        let _ = &self.client;
+        let _ = &self.pair;
+        let _ = &self.cf;
+        unimplemented!()
+    }
+}
+
+pub struct BatchPut<'a, AClient: Client> {
+    client: &'a AClient,
```

Comment:
This code doesn't compile:

```
error[E0309]: the parameter type `AClient` may not live long enough                          
   --> src/raw.rs:122:5                                                                      
    |                                                                                        
121 | pub struct BatchPut<'a, AClient: Client> {                                             
    |                         -------- help: consider adding an explicit lifetime bound `AClient: 'a`...                                                                                  
122 |     client: &'a AClient,                                                               
    |     ^^^^^^^^^^^^^^^^^^^                                                                
    |                                                                                        
note: ...so that the reference type `&'a AClient` does not outlive the data it points at     
   --> src/raw.rs:122:5                                                                      
    |                                                                                        
122 |     client: &'a AClient,                                                               
    |     ^^^^^^^^^^^^^^^^^^^          
```

There are a few others, let's make sure that the RFC we merge has building code. :)

Reply:
Oops. Forgot to use stable toolchain. It's fixed and should work now.

Reply:
:+1: